src/auth/auth.service.ts

```diff
@@ -50,8 +56,64 @@ export class AuthService {
             throw new UnauthorizedException("Wrong Credentials");
         }
 
-        return this.generateToken( user.id, user.username,);
+        return this.generateToken( user.id, user.username);
+
+    }
+
+    async changePassword(data: changePasswordDto, userId: number): Promise<any>{
+        if(!userId){
+            throw new BadRequestException("Data missing");
+        }
+        const user = await this.userService.findById(userId,  ['password']);
+   
+        if(!user){
+            throw new UnauthorizedException("Wrong Credentials");
+        }
+
+        const isPasswordMatched = await bcrypt.compare(data.oldPassword, user.password);
+
+        if(!isPasswordMatched){
+           
+            throw new UnauthorizedException("Wrong Credentials");
+        }
+
+        //hash new password
+        const newHashedPassword = await this.hashPassword(data.newPassword);
+        return this.userService.updatePassword(userId, newHashedPassword);
+    }
+
+    async forgotPassword(data: forgotPasswordDto): Promise<any>{
+        const user = await this.userService.findByEmail(data.email);
+
+        if(user){
```

Comment:
if user not found then return user not found

Reply:
This was done intentionally! While doing forgot password if any hacker is trying with different emails we don't want to tell user that "This user is not exists"